### PR TITLE
Relocate BigQuery Partition Utils

### PR DESCRIPTION
**Pull request by jayehwhyehentee**
Move `table/restrictions/BigQueryPartition` to `common/utils/BigQueryParititonUtils`

### CODE REVIEW

File: flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/common/utils/BigQueryPartitionUtilsTest.java [hunk 1]
```diff
@@ -310,33 +315,15 @@ public void testCheckPartitionCompletedMonth() {
...
         PartitionIdWithInfoAndStatus partitionWithInfoAndStatus =
-                BigQueryPartition.checkPartitionCompleted(partitionWithInfo);
+                BigQueryPartitionUtils.checkPartitionCompleted(partitionWithInfo);
 
         Assertions.assertThat(partitionWithInfoAndStatus.getStatus())
-                .isEqualTo(BigQueryPartition.PartitionStatus.COMPLETED);
-    }
-
-    @Test
-    public void testCheckPartitionNotCompletedMonth() {
```
Comment (vishalkarve15) [inline flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/common/utils/BigQueryPartitionUtilsTest.java:325]: This got deleted by mistake.

Comment (jayehwhyehentee) [reply flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/common/utils/BigQueryPartitionUtilsTest.java]: Thanks for catching. Restored

**Review by vishalkarve15 — APPROVED** [review]

**Pull request merged**